ledger/src/advance.rs

```diff
@@ -217,9 +228,23 @@ impl<N: Network, C: ConsensusStorage<N>> Ledger<N, C> {
                 // Retrieve the latest proof target.
                 let latest_proof_target = self.latest_proof_target();
                 // Separate the candidate solutions into valid and aborted solutions.
+                let mut accepted_solutions: IndexMap<Address<N>, u64> = IndexMap::new();
                 let (valid_candidate_solutions, aborted_candidate_solutions) =
                     split_candidate_solutions(candidate_solutions, N::MAX_SOLUTIONS, |solution| {
-                        self.puzzle().check_solution_mut(solution, latest_epoch_hash, latest_proof_target).is_ok()
+                        let prover_address = solution.address();
+                        let num_accepted_solutions = accepted_solutions.get(&prover_address).copied().unwrap_or(0);
+                        // Determine the the prover has reached their solution limit.
+                        if self.has_reached_solution_limit(&prover_address, num_accepted_solutions) {
```

Comment:
I think this might need to be `self.has_reached_solution_limit(&prover_address, num_accepted_solutions + 1)`

Let's say a prover is allowed to submit 2 solutions. The cache contains 1, `accepted_solutions` contains 1. This check will pass. And as a result, we will be adding **another** solution to `accepted_solutions`

Reply:
It should be correct as is. In your example of 2 allowed solutions in an epoch, if cache has 1 already, then there is only 1 allocation left.

If the incoming block has 2 more solutions. The first one will be accepted, which sets num_accepted_solutions to 1. The second solution in the block will now return true for `self.has_reached_solution_limit(&prover_address, num_accepted_solutions)` and be ignored.

Reply:
I stand corrected.

If you have the time and energy, a test for this with a couple different cases would be nice, e.g.:
- empty prover cache v.s. non-empty prover cache
- solution in block within limit, solution above limit